myia/prim/grad_implementations.py

```diff
@@ -0,0 +1,245 @@
+"""Implementations of the primitives' gradients.
+
+Each primitive is associated to an augmented function, which returns a pair of
+the (augmented) original primitive's output and a backpropagator function.
+"""
+
+from ..api import standard_pipeline
+from ..composite import zeros_like
+from ..info import NamedDebugInfo, About
+from ..ir import Constant, Graph, manage, clone, MetaGraph
+from ..utils import Registry, newenv
+
+from . import ops as primops
+from .py_implementations import \
+    Jinv, J, \
+    scalar_mul, scalar_div, scalar_sub, scalar_usub, scalar_log, scalar_pow, \
+    tuple_setitem, switch
+
+
+parse = standard_pipeline \
+    .select('parse') \
+    .make_transformer('input', 'graph')
+
+
+def bprop_to_augm(prim, fn):
+    """Given a function for the bprop, make the augmented function."""
+    info = NamedDebugInfo(prim=prim, name=prim.name)
+
+    bprop = parse(fn)
```

Comment:
Add a clone here, otherwise you'll get manager errors.  Also because you modify the returned graph.

Reply:
The `clone(outer)` at the end of the function will clone `bprop` along with it, but it's true that it's better to clone a graph before modifying it, so I'll fix that.

Reply:
OK.